src/Agent.Listener/DistributedTask.Pipelines/Yaml/Contracts/CheckoutStep.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.TeamFoundation.DistributedTask.Orchestration.Server.Pipelines.Yaml.TypeConverters;
+
+namespace Microsoft.TeamFoundation.DistributedTask.Orchestration.Server.Pipelines.Yaml.Contracts
+{
+    internal sealed class CheckoutStep : ISimpleStep
+    {
+        public String Name { get; set; }
+
+        internal String Clean { get; set; }
```

Comment:
Gotcha, this is just POCO transfer objects. The `Contracts` stuff threw me a bit.
I was more poking around to see how the Yaml serialization of jobs would look like as it ended up as an on-premise :smile: